src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -84,57 +90,46 @@ public class NuGetSearcherManager : SearcherManager<NuGetIndexSearcher>
             }
 
             LoggerFactory = loggerFactory;
+            FrameworkLogger logger = loggerFactory.CreateLogger<NuGetSearcherManager>();
 
-            NuGetSearcherManager searcherManager;
-            var luceneDirectory = configuration.Get("Local.Lucene.Directory");
-
+            // If a local Lucene directory has been specified, create a directory and loader for the specified directory.
+            var luceneDirectory = await settings.GetOrDefault<string>(IndexingSettings.LocalLuceneDirectory);
             if (!string.IsNullOrEmpty(luceneDirectory))
             {
-                string dataDirectory = configuration.Get("Local.Data.Directory");
-
-                searcherManager = new NuGetSearcherManager(
-                    luceneDirectory,
-                    loggerFactory.CreateLogger<NuGetSearcherManager>(),
-                    directory ?? new SimpleFSDirectory(new DirectoryInfo(luceneDirectory)),
-                    loader ?? new FileLoader(dataDirectory),
-                    azureDirectorySynchronizer: null);
+                directory = directory ?? new SimpleFSDirectory(new DirectoryInfo(luceneDirectory));
+                loader = loader ?? new FileLoader(await settings.GetOrDefault<string>(IndexingSettings.LocalDataDirectory));
+            }
+
+            IIndexDirectoryProvider indexProvider;
+            if (directory == null)
+            {
+                // If no directory has been provided, create an IndexDirectoryProvider from the configuration.
+                indexProvider = await IndexDirectoryProvider.Create(settings, logger);
             }
             else
             {
```

Comment:
Unfortunately, with the addition of the `IIndexDirectoryProvider` interface, splitting the scenarios would now have far more redundant code than before. I think I'll keep it as is but I'll take a look to see if I can make it easier to read.